ReactWindows/ReactNative/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,255 @@
+using ReactNative.UIManager;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Peers;
+
+namespace ReactNative.Common
+{
+    /// <summary>
+    /// Accessibility helper methods.
+    /// </summary>
+    public static class AccessibilityHelper
+    {
+        #region ImportantForAccessibility
+
+        /// <summary>
+        /// Initializes the value of AccessibilityView for <paramref name="uiElement"/> and its children,
+        /// based on the ancestor settings.
+        /// </summary>
+        /// <param name="uiElement"></param>
+        public static void InitImportantForAccessibility(UIElement uiElement)
+        {
+            var uiElementAutomationPeer = FrameworkElementAutomationPeer.FromElement(uiElement);
+            if (IsHiddenByAncestor(uiElementAutomationPeer))
+            {
+                // If an ancestor is "hiding" the element, set AccessibilityView to Raw.
+                AutomationProperties.SetAccessibilityView(uiElement, AccessibilityView.Raw);
+                SetChildrenAccessibilityView(uiElementAutomationPeer, AccessibilityView.Raw);
+            }
+        }
+
+        /// <summary>
+        /// Sets the ImportantForAccessibility property for <paramref name="uiElement"/>.
+        /// It uses AutomationProperties.AccessibilityView to implement the element and its children
+        /// visibility to narrator. ImportantForAccessibility value is stored as an attached property.
+        /// </summary>
+        /// <param name="uiElement"></param>
+        /// <param name="importantForAccessibilityNew"></param>
+        public static void SetImportantForAccessibility(UIElement uiElement, ImportantForAccessibility importantForAccessibilityNew)
+        {
+            var importantForAccessibilityOld = GetImportantForAccessibilityAttached(uiElement);
+            if (importantForAccessibilityOld == importantForAccessibilityNew)
+            {
+                return;
+            }
+
+            SetImportantForAccessibilityAttached(uiElement, importantForAccessibilityNew);
+
+            // Check if the element is "hidden" (i.e. AccessibilityView = 'Raw') by an ancestor. This occurs if
+            // there is an ancestor with 'Yes' or 'NoHideDescendants'. In this case make no change to AccessibilityView.
+            // If later the element or its parent ImportantForAccessibility is set to 'Auto' or 'No', that makes children
+            // visible, SetChildrenAccessibilityViewFromImportantForAccessibility will update AccessibilityView properly
+            // for the element and its children accordingly.
```

Comment:
As discussed in person, let's remove this comment and replace it with a description of why we can return a few lines down.

Reply:
Fixed.

---
In reply to: [164515357](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r164515357) [](ancestors = 164515357)